portal-kernel/src/com/liferay/portal/kernel/portlet/PortletFilterUtil.java

```diff
@@ -28,12 +30,22 @@
 import javax.portlet.ResourceRequest;
 import javax.portlet.ResourceResponse;
 import javax.portlet.filter.FilterChain;
+import javax.portlet.filter.HeaderFilterChain;
 
 /**
  * @author Michael Young
+ * @author Neil Griffin
  */
 public class PortletFilterUtil {
 
+	public static void doFilter(
```

Comment:
@ngriffin7a why this must be singled out? Can't we use lifecycle to indicate it is header phase just like others?

Reply:
@shuyangzhou It is singled out because javax.portlet.HeaderFilterChain https://github.com/apache/portals-pluto/blob/master/portlet-api/src/main/java/javax/portlet/filter/HeaderFilterChain.java#L44
and javax.portlet.FilterChain https://github.com/apache/portals-pluto/blob/master/portlet-api/src/main/java/javax/portlet/filter/FilterChain.java#L48 do not have any common inheritance. In other words, we can't pass an instance of HeaderFilterChain to a method that takes a FilterChain.